x-pack/plugins/monitoring/server/types.ts

```diff
@@ -79,10 +80,25 @@ export interface RouteDependencies {
   logger: Logger;
 }
 
+export type MonitoringRouteConfig<Params, Query, Body, Method extends RouteMethod> = {
+  // NOTE / TODO: These uppercase versions are here temporarily until all routes are converted to TS,
+  // and using the standard RouteMethod type.
+  method: RouteMethod | 'GET' | 'POST' | 'PUT';
+} & RouteConfig<Params, Query, Body, Method> & {
+    // NOTE / TODO: Ideally we'd make all routes stop using this nested custom "config" as
+    // validate already exists on the RouteConfig type.
+    config?: {
+      validate: RouteConfig<Params, Query, Body, Method>['validate'];
+    };
+    handler: (request: LegacyRequest) => any;
```

Comment:
Please see caveats of `LegacyRequest` and not being able to use the core `RequestHandler` type in the description.